### PR TITLE
fix: create chart from dashboard

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: 

### Description:

https://github.com/lightdash/lightdash/assets/962095/aa7a3a4b-feba-4c45-8ddc-a4808d40c0e6

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/hooks/dashboard/useDashboardStorage.tsx [hunk 1]
```diff
@@ -67,23 +67,14 @@ const useDashboardStorage = () => {
...
-        getHasDashboardChanges,
-    ]);
+    return {
+        storeDashboard: storeDashboard,
+        clearDashboardStorage: clearDashboardStorage,
+        getEditingDashboardInfo: getEditingDashboardInfo,
+        getIsEditingDashboardChart: getIsEditingDashboardChart,
+        clearIsEditingDashboardChart: clearIsEditingDashboardChart,
+        getHasDashboardChanges: getHasDashboardChanges,
+    };
```
Comment (IrakliJani) [inline packages/frontend/src/hooks/dashboard/useDashboardStorage.tsx:77]: no need to double memo this

File: packages/frontend/src/hooks/toaster/useToaster.tsx [hunk 2]
```diff
@@ -90,10 +90,6 @@ const useToaster = () => {
                             )}
                         </Stack>
                     ) : undefined,
-                onOpen: (props: NotificationProps) => {
-                    rest.onOpen?.(props);
-                    if (props.id) openedKeys.current.add(props.id);
-                },
```
Comment (IrakliJani) [inline packages/frontend/src/hooks/toaster/useToaster.tsx:96]: raised some warnings in the console. mantine does not have that.  + we don't use this anywhere in the codebase

File: packages/frontend/src/pages/Dashboard.tsx [hunk 3]
```diff
@@ -201,50 +202,46 @@ const Dashboard: FC = () => {
...
-            clearIsEditingDashboardChart();
-            const unsavedDashboardTilesRaw = sessionStorage.getItem(
-                'unsavedDashboardTiles',
+        if (isDashboardLoading) return;
+        if (dashboardTiles) return;
+
+        setDashboardTiles(dashboard?.tiles ?? []);
+    }, [isDashboardLoading, dashboard, dashboardTiles, setDashboardTiles]);
+
+    useEffect(() => {
```
Comment (IrakliJani) [inline packages/frontend/src/pages/Dashboard.tsx:212]: This `useEffect` was quite complex; it was firing unpredictably and often doubled. I've made it a bit nicer, but we should refactor the entire "create chart from dashboard" flow at some point to make it less error-prone.

**Review by joaoviana — APPROVED** [review]
Tested the following: 
* Create new dashboard
* Add saved charts (with remove and re-add, and refresh)
* Add new chart (from dashboard) - tested with saving but also with discarding changes 
* Temp and saved filters
* dashboard export with filters + overrides

File: packages/frontend/src/pages/Dashboard.tsx (hunk 3 above)
Comment (magnew) [reply packages/frontend/src/pages/Dashboard.tsx:212]: Definitely.

**Pull request merged**